WalletWasabi/WabiSabi/Models/MultipartyTransaction/SigningState.cs

```diff
@@ -72,7 +83,7 @@ public Transaction CreateUnsignedTransaction()
 
 			foreach (var txout in Outputs)
 			{
-				tx.Outputs.AddWithOptimize(txout.Value, txout.ScriptPubKey);
+				tx.Outputs.Add(txout);
```

Comment:
oops, I missed this change I think (or did it come later?)

FWIW, I always felt like `AddWithOptimize` was a bit dangerous, since it introduces some ambiguity.

That said if there are two identical outputs, a signer can't really tell which is theirs, so the coordinator could in principle only add one of those and obtain signatures from both (in the case that say two users are donating to the same public address).

The only solution here is to validate the *transcript* of the multiparty transaction state updates (as opposed to the final state), which we don't do yet but which I feel very strongly that we should before mainnet. And from this point of view using `Add` instead of `AddWithOptimize` is much much easier to analyze.